src/components/SchemaRow/useChoices.ts

```diff
@@ -0,0 +1,89 @@
+import { pointerToPath } from '@stoplight/json';
+import { isReferenceNode, isRegularNode, SchemaNode } from '@stoplight/json-schema-tree';
+import { last } from 'lodash';
+import * as React from 'react';
+
+import { isComplexArray, isNonEmptyParentNode } from '../../tree';
+import { printName } from '../../utils';
+
+type Choice = {
+  title: string;
+  type: SchemaNode;
+};
+
+function calculateChoiceTitle(node: SchemaNode, isPlural: boolean): string {
+  const complexObjectSuffix = isPlural ? '-s' : '';
+  const primitiveSuffix = isPlural ? 's' : '';
+  if (isRegularNode(node)) {
+    const realName = printName(node);
+    if (realName) {
+      return realName + complexObjectSuffix;
+    }
+    return node.primaryType !== null ? node.primaryType + primitiveSuffix : 'any';
+  }
+  if (isReferenceNode(node)) {
+    const lastPiece = last(pointerToPath(node.value ?? ''));
```

Comment:
```suggestion
    const lastPiece = !node.error && node.value ? last(pointerToPath(node.value)) : null;
```
`pointerToPath` will throw an exception if invalid $ref is given.